starknet_py/contract.py

```diff
@@ -460,6 +460,12 @@ async def deploy(
         :raises: `ValueError` if neither compilation_source nor compiled_contract is provided.
         :return: DeployResult instance
         """
+        warnings.warn(
+            "In the future versions of StarkNet, Deploy transaction will not be supported."
+            "To deploy a contract use cairo syscall or Universal Deployer Contract",
+            category=DeprecationWarning,
+        )
```

Comment:
I'm not sure if we should be mentioning the UDC just yet. Maybe let's include it in the warning once it's deployed on testnet and our support is released.

We can create issue to do this.